metrics/ci_integrations/test/client/buildkite/buildkite_client_test.dart

```diff
@@ -398,10 +399,66 @@ void main() {
       },
     );
 
+    test(
+      ".fetchBuild() throws an ArgumentError if the given pipeline slug is null",
+      () async {
+        expect(() => client.fetchBuild(null, buildNumber), throwsArgumentError);
```

Comment:
```suggestion
        final result = client.fetchBuild(null, buildNumber);
        expect(result, throwsArgumentError);
```
Would it work as the `result` is the`Future`? If yes, then this seems to be more readable.